components/sup/src/manager/service/config.rs

```diff
@@ -81,12 +82,17 @@ impl ServiceConfig {
     pub fn new(package: &PackageInstall,
                mgr_cfg: &ManagerConfig,
                runtime_cfg: &RuntimeConfig,
-               config_root: Option<PathBuf>,
+               config_root_from_config: Option<PathBuf>,
                bindings: Vec<(String, ServiceGroup)>)
                -> Result<ServiceConfig> {
-        let config_root = config_root.unwrap_or(package.installed_path.clone());
+        let config_root = match config_root_from_config {
+            Some(ref root) => root.clone(),
+            None => package.installed_path.clone(),
+        };
         Ok(ServiceConfig {
-            pkg: Pkg::new(package, runtime_cfg)?,
+            pkg: Pkg::new(package,
```

Comment:
I think we do want to pass a ref to the runtime config and not just the user/group.

We should totally be storing a ref to the RuntimeConfig and not just pull it's parts out in the Pkg constructor like we previously had, but that refactor/optimization hasn't happened yet. This is in line with the spirit of this PR - don't duplicate memory into caches that get stale ;)